src/test/java/com/stripe/functional/BankAccountTest.java

```diff
@@ -135,9 +135,14 @@ public void testDelete() throws StripeException {
     final Customer customer = Customer.retrieve(CUSTOMER_ID);
     final BankAccount bankAccount = getBankAccountFixture(customer);
 
-    final DeletedBankAccount deletedBankAccount = bankAccount.delete();
+    final BankAccount deletedBankAccount = bankAccount.delete();
 
     assertNotNull(deletedBankAccount);
+    /**
+     * TODO(spastorelli): stripe-mock currently doesn't include deleted:true in the response of a
+     * DELETE  /v1/customers/cus_1234/sources/* call. Uncomment below assertion when it does.
+     */
+    //assertTrue(deletedBankAccount.getDeleted());
```

Comment:
@brandur-stripe Do you have a sense of how difficult it would be to handle this correctly in stripe-mock?

Reply:
So the basic problem here is that the endpoint sometimes returns non-deleted objects (I believe deleting a source "detaches" it, but it's still a live object):

```
      responses:
        '200':
          content:
            application/json:
              schema:
                anyOf:
                - "$ref": "#/components/schemas/customer_source"
                - "$ref": "#/components/schemas/deleted_bank_account"
                - "$ref": "#/components/schemas/deleted_object"
                - "$ref": "#/components/schemas/deleted_card"
```

So stripe-mock is just doing the dumbest thing: picking the first branch, and returning that. If it had picked any of the last three, there would be a `deleted` field.

The only good fix I can think of is to try and recognize the resource type based on the input ID, and that'll require a little bit of work.

Reply:
Ack. @spastorelli-stripe I recommend that you stub this request as well for now.

Reply:
Thanks @ob-stripe and @brandur-stripe for your input and suggestions. I've stubbed the request.